lib/puppet/type/file/selcontext.rb

```diff
@@ -0,0 +1,153 @@
+# Manage SELinux context of files.
```

Comment:
@joshcooper should I be pulling this file as well, or no? k5login uses it (which will be a dependency we can specify when we pull k5login out), but it's also loaded as a part of the file type

Reply:
@branan @joshcooper was saying you had thoughts on selinux extraction? Is this a file we can move out to the module, or should we keep this in the puppet repo?

Reply:
Thisi needs to stay in Puppet - it's related to how the `file` type handles selinux permissions on files. Some selinux bits for core types will always have to stay in Puppet itself (until we figure out how to modularize the really core types)

I'm also /kind/ of inclined to leave the `feature` in core, since it's super lightweight and might be useful when adding selinux awareness to other types/providers. But I don't totally have a use-case yet, so as long as bringing it back later won't break anything, I'm fine pulling it out for now.

Reply:
👍 sweet, I'll pull those out